compliance/http/src/test/java/org/eclipse/rdf4j/http/server/ProtocolTest.java

```diff
@@ -197,6 +203,31 @@ public void testUpdateDirect_POST()
 	}
 
 	/**
+	 * Checks that the server accepts a formencoded POST with an update and a timeout parameter.
+	 */
+	@Test
+	public void testUpdateForm_POST()
+		throws Exception
+	{
+		String update = "delete where { <monkey:pod> ?p ?o . }";
+		String location = Protocol.getStatementsLocation(TestServer.REPOSITORY_URL);
+		CloseableHttpClient httpclient = HttpClients.createDefault();
+		HttpPost post = new HttpPost(location);
+		List<NameValuePair> nvps = new ArrayList<NameValuePair>();
+		nvps.add(new BasicNameValuePair(Protocol.UPDATE_PARAM_NAME, update));
+		nvps.add(new BasicNameValuePair(Protocol.TIMEOUT_PARAM_NAME, "1"));
+		UrlEncodedFormEntity entity = new UrlEncodedFormEntity(nvps, Charsets.UTF_8);
+
+		post.setEntity(entity);
+
+		CloseableHttpResponse response = httpclient.execute(post);
+
+		System.out.println("Update Form Post Status: " + response.getStatusLine());
+		int statusCode = response.getStatusLine().getStatusCode();
+		assertEquals(true, statusCode >= 200 && statusCode < 400);
+	}
+
+	/**
```

Comment:
This requires that we include spring-test (which contains ready-to-use Mock classes for HttpServletRequest etc) as a dependency. Not a problem, but I need to file a piggyback CQ for this ( https://dev.eclipse.org/ipzilla/show_bug.cgi?id=11653 ). Given that it's already used by other Eclipse projects approval of this CQ should be swift.